ast/desugar/Desugar.cc

```diff
@@ -929,19 +944,28 @@ ExpressionPtr node2TreeImplBody(DesugarContext dctx, parser::Node *what) {
                     DuplicateHashKeyCheck::checkSendArgs(dctx, numPosArgs, args);
 
                     ExpressionPtr res;
-                    if (block == nullptr) {
+                    if (blockPassArg == nullptr) {
+                        // Desugar any call without a block pass argument. Block literals are handled here.
+                        // E.g. `a.map { |x| x.to_s }
                         res = MK::Send(loc, move(rec), send->method, send->methodLoc, numPosArgs, move(args), flags);
                     } else {
-                        if (auto lit = cast_tree<Literal>(block); lit && lit->isSymbol()) {
+                        if (auto lit = cast_tree<Literal>(blockPassArg); lit && lit->isSymbol()) {
+                            // Desugar a call with a Symbol block pass argument, like `a.map(&:to_s)`
```

Comment:
IMO, the real subtlety here is the in how the symbol2Proc function works. It uses tuples and there's custom support for type inference on tuples in calls.cc that powers this case.

IMO, this is one of the only cases worth documenting here because of how subtle it is.

Reply:
I added it on symbol2Proc.